src/main/java/bio/terra/cloudfunctions/common/App.java

```diff
@@ -0,0 +1,33 @@
+package bio.terra.cloudfunctions.common;
+
+import bio.terra.cloudevents.CloudStorageEventType;
+import bio.terra.cloudevents.GCSEvent;
+
+/**
+ * This abstract class represents the business application that interprets CloudEvent messages.
+ *
+ * <p>Sub-classes overrides the process() method with business logic to achieve desired functional
+ * goals.
+ *
+ * <p>Sub-classes can be integrated with a DI framework to deploy the business logic like as a
+ * Service.
+ */
+public abstract class App {
+  private CloudStorageEventType cloudStorageEventType;
+  private GCSEvent cloudStorageEvent;
```

Comment:
I think we're too tightly coupled to Google when we use `GCSEvent` here.  We should use our own class that contains the necessary info to download the file and to detect duplicate messages.

Reply:
Updated `App` to use `FileUploadedMessage`.